src/robusta/cli/auth.py

```diff
@@ -0,0 +1,156 @@
+from cryptography.hazmat.primitives.asymmetric import rsa
+from pydantic import BaseModel
+
+from .backend_profile import backend_profile
+from .playbooks_cmd import NAMESPACE_EXPLANATION, get_playbooks_config
+from .utils import namespace_to_kubectl
+
+AUTH_SECRET_NAME = "robusta-auth-config-secret"
+app = typer.Typer()
+
+
+class RSAKeyPair(BaseModel):
+    prv: str
+    pub: str
+
+
+def gen_rsa_pair() -> RSAKeyPair:
+    # generate private/public key pair
+    key = rsa.generate_private_key(public_exponent=65537, key_size=2048)
+
+    # get public key in OpenSSH format
+    public_key = key.public_key().public_bytes(Encoding.PEM, PublicFormat.SubjectPublicKeyInfo)
+
+    # get private key in PEM container format
+    pem = key.private_bytes(
+        encoding=Encoding.PEM,
+        format=PrivateFormat.TraditionalOpenSSL,
+        encryption_algorithm=NoEncryption()
+    )
+
+    return RSAKeyPair(
+        pub=public_key.decode('utf-8'),
+        prv=pem.decode('utf-8')
+    )
+
+
+def get_auth_config(namespace: str) -> Optional[RSAKeyPair]:
+    try:
+        secret_content = subprocess.check_output(
+            f"kubectl get secret {namespace_to_kubectl(namespace)} {AUTH_SECRET_NAME} -o yaml",
+            shell=True,
+        )
+    except Exception:
+        return None
+
+    auth_secret = yaml.safe_load(secret_content)
+    return RSAKeyPair(
+        prv=base64.b64decode(auth_secret["data"]["prv"]).decode(),
+        pub=base64.b64decode(auth_secret["data"]["pub"]).decode()
+    )
+
+
+class TokenDetails(BaseModel):
+    pub: str
+    account_id: str
+    user_id: str
+    session_token: str
+    enc_key: str
+    key_id: str
+
+
+def store_server_token(token_details: TokenDetails, debug: bool = False) -> bool:
+    try:
+        response = requests.post(backend_profile.robusta_store_token_url, json=token_details.dict())
+        if debug and response.status_code != 201:
+            typer.secho(f"Failed to store server token. status-code {response.status_code} text {response.text}")
+
+        return response.status_code == 201
+    except Exception as e:
+        if debug:
+            typer.secho(f"Error trying to store server token. {traceback.format_exc()}")
+        return False
+
+
+@app.command()
+def gen_token(
+    account_id: str = typer.Option(
+        None,
+        help="Robusta account id",
+    ),
+    user_id: str = typer.Option(
+        None,
+        help="User id for which the token is created",
+    ),
+    session_token: str = typer.Option(
+        None,
+        help="User session token. Created for an authenticated user via the Robusta UI",
+    ),
+    namespace: str = typer.Option(
+        None,
+        help=NAMESPACE_EXPLANATION,
+    ),
+    debug: bool = typer.Option(False),
+):
+    """Generate token required to run actions manually in Robusta UI"""
+    if not account_id or not user_id or not session_token:
```

Comment:
You can get typer to handle this check for you. https://typer.tiangolo.com/tutorial/options/required/

Reply:
fixed